tools/src/test/scala/scala/scalanative/NIRCompilerTest.scala

```diff
@@ -68,6 +68,36 @@ class NIRCompilerTest extends AnyFlatSpec with Matchers with Inspectors {
       NIRCompiler(_.compile(code))
     }
   }
+  it should "report error for extern in val definition" in {
+    // given
+    val code =
+      """import scala.scalanative.unsafe.extern
+        |
+        |@extern
+        |object Dummy {
+        |  val foo: Int = extern
+        |}""".stripMargin
+    // when
+    val caught = intercept[CompilationFailedException] {
+      NIRCompiler(_.compile(code))
+    }
+    caught.getMessage() should include(
+      "`extern` cannot be used in val definition"
+    )
+  }
+
+  it should "compile extern var definition" in {
+    // given
+    val code =
+      """import scala.scalanative.unsafe.extern
+        |
+        |@extern
+        |object Dummy {
+        |  var foo: Int = extern
+        |}""".stripMargin
+    // when
+    NIRCompiler(_.compile(code))
+  }
```

Comment:
Interesting, this is allowed? What can it be used for?

Reply:
Ah, JVM type in extern variable does not make sense. I'm not sure whether scala native implicitly convert JVM type to corresponding c type.

If  your｀this｀ means extern variables in extern object, there are several usecases.
https://github.com/scala-native/scala-native/blob/main/posixlib/src/main/scala/scala/scalanative/posix/getopt.scala

Reply:
Hmm, not sure I understand your comment. I didn't mean about `this`, I mean about `var`. I don't think a `var` can be used for `extern`.

Reply:
scala-native itself uses var for extern.

```scala
package scala.scalanative
package posix

import scalanative.unsafe._

@extern
object getopt {
  var optarg: CString = extern
```

Reply:
Interesting, I did not know that. Thanks! :)